File: Week2/node/utils/parse.js
function parseNumberList(input) {
  const tokens = input.trim().split(/[,\s]+/).filter(Boolean);

  if (tokens.length === 0) {
    return { ok: false, error: "You did not enter any numbers." };
  }

  const numbers = tokens.map((t) => Number(t));
  if (numbers.some((n) => Number.isNaN(n))) {
    return { ok: false, error: "Invalid input: please enter only numbers." };
  }

  return { ok: true, value: numbers };
}

function parseNumber(input) {
  const n = Number(input.trim());
  if (Number.isNaN(n)) {
    return { ok: false, error: "Please enter a valid number." };
  }
  return { ok: true, value: n };
}

module.exports = { parseNumberList, parseNumber };

File: Week2/python/utils/parse.py
def parse_number_list(raw: str):
    text = raw.strip()
    if not text:
        return {"ok": False, "error": "You did not enter any numbers."}

    # supports spaces and commas
    parts = [p for p in text.replace(",", " ").split() if p]

    try:
        nums = [int(p) for p in parts]
        return {"ok": True, "value": nums}
    except ValueError:
        return {"ok": False, "error": "Invalid input: please enter only integers."}


def parse_number(raw: str):
    text = raw.strip()
    try:
        return {"ok": True, "value": int(text)}
    except ValueError:
        return {"ok": False, "error": "Please enter a valid integer."}
